RealmSwift/Object.swift

```diff
@@ -540,3 +540,50 @@ extension Object: AssistedObjectiveCBridgeable {
         return (objectiveCValue: unsafeCastToRLMObject(), metadata: nil)
     }
 }
+
+extension ObjectBase {
+
+    // MARK: Key Path Strings
+
+    /**
+     Gets the components of a given key path as a string.
+
+     - warning: Objects that declare properties with the old `@objc dynamic` syntax are not fully supported
+     by this function, and it is recommened that you use `@Persisted` to declare your properties if you wish to use
+     this function to its full benefit.
+
+     Example:
+     ```
+     let name = ObjectBase._name(for: \Person.dogs[0].name) // "dogs.name"
+     // Note that the above KeyPath expression is only supported with properties declared
+     // with `@Persisted`.
+     let nested = ObjectBase._name(for: \Person.address.city.zip) // "address.city.zip"
+     ```
+     */
+    public static func _name<T: ObjectBase>(for keyPath: PartialKeyPath<T>) -> String {
```

Comment:
Any particular reason this is public? It's not part of any protocol and if it is just for testing, that is not a good reason.

Reply:
Making it internal and testing it via @testable import means that we can't test it in release builds, and I could imagine this having ARC lifetime issues with optimizations. I guess it doesn't actually need to be on ObjectBase, though, so maybe it should be a free function?

Reply:
I acquiesce. As discussed offline, @leemaguire will make it a free function and we will keep it public.